### ISSUE TITLE
Profiling messenger in Datadog seems off

### ISSUE THREAD

**Issue by clemherreman** [sourceability/instrumentation#9]
Hi :)

I'm struggling to give meaning to how data goes to Datadog. Traces & spans seems to mix the `messenger:consume` command, and the individual consumed messages

Here's an example :

![image](https://user-images.githubusercontent.com/272812/159325890-6b3d2163-21dc-42c2-9b8c-6602d4f37623.png)

You can see individual messages around a ~2-5s, but the first consumed one is always -5-7minutes, which is likely to be the worker process restart time (10 min),

My config is the following:

```yaml
# config/sourceability_instrumentation.yaml
sourceability_instrumentation:
  profilers:
    datadog:
      enabled: true

  listeners:
    command:
      enabled: true

    messenger:
      enabled: true
```

```yaml
#config/messenger.yaml
framework:
    messenger:
        buses:
            messenger.bus.default:
                middleware:
                    - Sourceability\Instrumentation\Messenger\ProfilerMiddleware
```

Am I doing something wrong? Are the messenger listener & messenger middlewares stepping on each others toes?

**Comment by clemherreman** [comment]
Sorry, I missclicked while writing this issue. Will open a proper one.

**Comment by adrienbrault** [comment]
Hey 👋

I see in your examples that you have both `sourceability_instrumentation.listeners.messenger.enabled: true` and the `Sourceability\Instrumentation\Messenger\ProfilerMiddleware`.


I've always used **either** of those, not **both**, they aren't meant to be complementary.
Most recently I've used the messenger middleware with `sourceability_instrumentation.listeners.messenger.enabled: false`.

I'd suggest trying to disable either and see what changes.

---

Going [through the code](https://github.com/sourceability/instrumentation/blob/64cab9c8f2cb11b5cc6cf6ca27035a40e7cb063f/src/Profiler/DatadogProfiler.php#L36-L46), I see that the following environment variables are expected:
```
DD_TRACE_GENERATE_ROOT_SPAN=0
DD_TRACE_AUTO_FLUSH_ENABLED=1
```

Do you have the same configuration? If not, could you try configuring the agent like this?

---

I haven't used this part of the library for a while, so I'm not easily able to reproduce. I would love for your feedback on the above suggestions, I think there's several improvements I could make to the README. PRs always welcome of course :)

**Comment by clemherreman** [comment]
Hi @adrienbrault :wave: 

Indeed, I was using **both**, with even the `command` listener :smile: 

Switching to only the handler fixed it, with some blind switching things I ended up guessing that :)

I see that the differences between the listener and the handler are the way those are called (*in messenger* vs *via the event dispatcher*), aside from that, are there differences? Which one is "better"?

<hr> 

I will make a PR to reflect that in the README, encouraging users to chose between the listener and the messenger handler :)
Do you mind explaining why you are using one vs the other?

Thanks for you time (and your lib)

**Comment by adrienbrault** [comment]
At @golem-ai we use the middleware, not the listener.

I honestly can't remember which one is better and why 😬

**Comment by clemherreman** [comment]
Closing, as this should be fixed by #10 :)
